front/pages/api/w/[wId]/vaults/[vId]/data_sources/index.ts

```diff
@@ -0,0 +1,542 @@
+            message:
+              "Could not create a system API key for the managed data source.",
+          },
+        });
+      }
+
+      const dustProject = await coreAPI.createProject();
+      if (dustProject.isErr()) {
+        return apiError(req, res, {
+          status_code: 500,
+          api_error: {
+            type: "internal_server_error",
+            message: `Failed to create internal project for the data source.`,
+            data_source_error: dustProject.error,
+          },
+        });
+      }
+
+      const dustDataSource = await coreAPI.createDataSource({
+        projectId: dustProject.value.project.project_id.toString(),
+        dataSourceId: dataSourceName,
+        config: {
+          embedder_config: {
+            embedder: {
+              max_chunk_size: embedderConfig.max_chunk_size,
+              model_id: embedderConfig.model_id,
+              provider_id: embedderConfig.provider_id,
+              splitter_id: embedderConfig.splitter_id,
+            },
+          },
+          qdrant_config: {
+            cluster: DEFAULT_QDRANT_CLUSTER,
+            shadow_write_cluster: null,
+          },
+        },
+        credentials: dustManagedCredentials(),
+      });
+
+      if (dustDataSource.isErr()) {
+        return apiError(req, res, {
+          status_code: 500,
+          api_error: {
+            type: "internal_server_error",
+            message: "Failed to create the data source.",
+            data_source_error: dustDataSource.error,
+          },
+        });
+      }
+
+      const dataSource = await DataSourceResource.makeNew(
+        {
+          assistantDefaultSelected:
+            isConnectorProviderAssistantDefaultSelected(provider),
+          connectorProvider: provider,
+          description: dataSourceDescription,
+          dustAPIProjectId: dustProject.value.project.project_id.toString(),
+          editedByUserId: user.id,
+          name: dataSourceName,
+          workspaceId: owner.id,
+        },
+        vault
+      );
+
+      // For each data source, we create two views:
+      // - One default view in its associated vault
+      // - If the data source resides in the system vault, we also create a custom view in the global vault until vault are released.
+      if (dataSource.vault.isSystem()) {
+        const globalVault = await VaultResource.fetchWorkspaceGlobalVault(auth);
+
+        await DataSourceViewResource.createViewInVaultFromDataSourceIncludingAllDocuments(
+          globalVault,
+          dataSource,
+          "custom"
+        );
+      }
+
+      const dataSourceView =
+        await DataSourceViewResource.createViewInVaultFromDataSourceIncludingAllDocuments(
+          dataSource.vault,
+          dataSource
+        );
+
+      const connectorsAPI = new ConnectorsAPI(
+        config.getConnectorsAPIConfig(),
+        logger
+      );
+
+      const connectorsRes = await connectorsAPI.createConnector(
```

Comment:
This endpoint logic is bit hard to read. Reading the code this looks like it's on the path of non connector data sources?

Reply:
Actually we're indeed in the connector path but it's hard to realize that. Let's maybe split in two functions to make things a bit clearer?

Reply:
We're indeed handling both managed and static ds. For the former route (not under vault), we had these two: 
https://github.com/dust-tt/dust/blob/main/front/pages/api/w/%5BwId%5D/data_sources/managed.ts
https://github.com/dust-tt/dust/blob/main/front/pages/api/w/%5BwId%5D/data_sources/index.ts

-> We quickly discussed it IRL last week and decided to have only one route now but I can split them again if it was better.

Reply:
done!